tests/publiccloud/xfsprepare.pm

```diff
@@ -117,34 +117,61 @@ sub create_config {
     assert_script_run("echo 'MKFS_OPTIONS=\"$mkfs_options\"' >> $CONFIG_FILE") unless ($mkfs_options eq '');
 }
 
-# Get all disks and partitions and return the first device, which has no parititons
-sub get_unused_nvme_device {
+# Get all disks and partitions and return the first device, which has no partitons
+# If the $dev_size argument is given, require the device to match the size aswell
+sub get_unused_device {
+    my ($dev_size) = @_;    # in GB
     my @disks = split /\n/, script_output('lsblk | grep disk | cut -d " " -f1');
+    my @sizes = split /\n/, script_output("lsblk -b | grep disk | awk '{print \$4}'");
     my @parts = split /\n/, script_output('lsblk | grep part | cut -d " " -f1');
-    foreach my $disk (@disks) {
-        # smartmatch is still experimental, but this can be enabled in the near future.
-        #return "/dev/$disk" unless ($disk ~~ @parts);
-        return "/dev/$disk" unless (grep { m/$disk/ } @parts);
+    for (my $i = 0; $i < @disks; $i++) {
```

Comment:
For my info, pls, is there any preference in your changed for-loop to indexed style? It is required much more attention and complexity to deal with each indexed value, instead of the already resolved value `$disk` as before.